arbnode/batch_poster.go

```diff
@@ -477,7 +486,18 @@ func (b *BatchPoster) checkReverts(ctx context.Context, to int64) (bool, error)
 					if shouldHalt {
 						logLevel = log.Error
 					}
-					logLevel("Transaction from batch poster reverted", "nonce", tx.Nonce, "txHash", tx.Hash, "blockNumber", r.BlockNumber, "blockHash", r.BlockHash)
+					txErr := arbutil.DetailTxErrorUsingCallMsg(ctx, b.l1Reader.Client(), tx.Hash, r, ethereum.CallMsg{
+						From:       tx.From,
+						To:         tx.To,
+						Gas:        uint64(tx.Gas),
+						GasPrice:   tx.GasPrice.ToInt(),
+						GasFeeCap:  tx.GasFeeCap.ToInt(),
+						GasTipCap:  tx.GasTipCap.ToInt(),
```

Comment:
Does this work if these are nil?

Reply:
Yep, go allows calling of methods (such as type conversion) on variables even if they are nil :)

Reply:
Ah, cool, didn't realize Big.ToInt() was nothing more than a conversion inside

```
// ToInt converts b to a big.Int.
func (b *Big) ToInt() *big.Int {
	return (*big.Int)(b)
}
```